replica/server.go

```diff
@@ -354,25 +380,60 @@ func (s *Server) Restore(url, name string) (err error) {
 		return err
 	}
 
-	if err := backupstore.RestoreDeltaBlockBackup(backupURL, toFile); err != nil {
-		return err
-	}
+	restoreObj := NewRestore(toFile)
 
-	newDisk := disk{
-		Parent:      "",
-		Name:        snapName,
-		Removed:     false,
-		UserCreated: false,
-		Created:     util.Now(),
+	config := &backupstore.DeltaRestoreConfig{
+		BackupURL: backupURL,
+		DeltaOps:  restoreObj,
+		Filename:  toFile,
 	}
 
-	if err := r.encodeToFile(newDisk, snapName+metadataSuffix); err != nil {
+	if err := backupstore.RestoreDeltaBlockBackup(config); err != nil {
```

Comment:
The restoration cannot be done in the replica since it's not in the data path. It should be done in the SyncAgentServer.

The sync agent server would need to call the replica server to perform revert. 

This restore function is for replica command restoration only, so it should be removed when this is merged in.